lib/pubid_iso/identifier.rb

```diff
@@ -0,0 +1,46 @@
+module PubidIso
+  class Identifier
+    STAGES = { NP: 10,
+               WD: 20,
+               CD: 30,
+               DIS: 40,
+               FDIS: 50,
+               PRF: 50,
+               IS: 60,
+               Fpr: 50 }.freeze
+
+    attr_accessor :number, :publisher, :copublisher, :stage, :substage, :part,
+                  :type, :year, :edition, :iteration, :supplements, :language
+
+    def initialize(stage: nil, supplements: nil, **opts)
+      if stage
+        @stage = STAGES[stage.to_sym]
+        @substage = 0
+      end
+
+      # XXX: temporary hack for ISO/IEC 19794-7:2014/Amd 1:2015/CD Cor 1 parsing
```

Comment:
@mico you may need to update the way parsing works for Amd and Cor. A Cor can be applied to an Amd document or a normal document. An Amd can apply on a normal document.

Reply:
This is now in #3